library/src/main/java/io/ona/kujaku/activities/MapActivity.java

```diff
@@ -371,25 +419,30 @@ public void showAlertDialog(int title, int message,
     @Override
     protected void onResume() {
         super.onResume();
-        if (mapView != null) mapView.onResume();
+        List<Point> droppedPoints = null;
+        if (kujakuMapView != null) {
+            kujakuMapView.onResume();
+            droppedPoints = kujakuMapView.getDroppedPoints();
```

Comment:
Could you expound on what flow of actions could lead to a duplication?

Reply:
Note that dropped points are maintained as a Set in KujakuMapView.

Thanks @githengi  for input on this.

Reply:
Does the MapView maintain state? Why drop points in `onResume` and at the same time at some point from `onCreate`? My assumption is that `onResume` can be called multiple times while the Activity has not yet been destroyed. At the same time, both methods are called during the same cycle of the Activity duplicating the calls

Reply:
The MapView does maintain state which avoids the duplication of points by maintaining a Set of points already dropped.

Is this an issue in the most recently updated code?

See if this line still exists.

Reply:
Does the MapView UI from Mapbox `com.mapbox.mapboxsdk.maps.MapView` maintain the dropped points in it's UI(rendered points) across different lifecycle states? Why do we keep on updating this in `onResume`?

Reply:
> My assumption is that onResume can be called multiple times while the Activity has not yet been destroyed

As according to https://developer.android.com/guide/components/activities/activity-lifecycle

Move it to `onCreate` and remove any redundant code used in checking for duplication

Reply:
Moved to onCreate.

Reply:
Code that checks for duplicates is still useful.

Reply:
Why would there be duplicates? `onCreate` is only called once and the `updateDroppedPoints` method is only used in the MapActivity where you push the points the first time

Reply:
Well, this is true for our MapActivity.

Consider a situation where another activity uses KujakuMapView to drop points and does not really validate for duplicates.

With the current implementation, all they would need to do is extend the Point class and override the `hashCode` method to provide their custom uniqueness constraints and not worry about duplicating points.

Is there a specific downside to having this functionality here that you can provide?

Reply:
OK, can you then not have two loops [here](https://github.com/onaio/kujaku/blob/133-add-drop-point-functionality-to-map-activity/library/src/main/java/io/ona/kujaku/views/KujakuMapView.java#L617) and [here](https://github.com/onaio/kujaku/blob/133-add-drop-point-functionality-to-map-activity/library/src/main/java/io/ona/kujaku/views/KujakuMapView.java#L625) and just have one loop? Yes, no?

Reply:
Cool, then why don't we defer the dropping until the mapbox Map has fully initialized. Current implementation means even if we are in the second loop and the mapboxMap has not yet fully initialized the points will never be dropped after that. Right?

Reply:
The points would still be dropped, since the global variable that stores dropped points is set.

The update method ideally takes care of updates after the map is initialized.

Before that, the `onMapReady` callback method will do the dropping once after initialization.

Reply:
Merged into one. Good discussion!